From 59eb7bd5c24cfea43265f90a486574a7e1c2654c Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Wed, 1 Feb 2023 13:59:57 +0000
Subject: [PATCH] Fix build with gcc-13

:Release Notes:
Add missing include.

:Detailed Notes:
Fixes:
http://gecko.lge.com/Errors/Details/529445
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/./base/types.h:47:3: error: 'uint64_t' does not name a type
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/./base/types.h:50:14: error: class 'gmp::base::stream_info_t' does not have any field named 'bit_rate'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/./base/types.h:54:3: error: 'uint32_t' does not name a type
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/./base/types.h:55:3: error: 'uint32_t' does not name a type
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/./base/types.h:59:14: error: class 'gmp::base::video_info_t' does not have any field named 'height'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/./base/types.h:59:4: error: class 'gmp::base::video_info_t' does not have any field named 'width'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/./base/types.h:65:3: error: 'uint32_t' does not name a type
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/./base/types.h:68:17: error: class 'gmp::base::audio_info_t' does not have any field named 'sample_rate'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/./base/types.h:72:3: error: 'uint32_t' does not name a type
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/./base/types.h:73:3: error: 'uint32_t' does not name a type
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/mediaresource/requestor.cpp:470:45: error: 'struct gmp::base::video_info_t' has no member named 'width'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/mediaresource/requestor.cpp:471:46: error: 'struct gmp::base::video_info_t' has no member named 'height'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:32:50: error: 'const struct gmp::base::video_info_t' has no member named 'bit_rate'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:33:48: error: 'const struct gmp::base::video_info_t' has no member named 'width'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:34:49: error: 'const struct gmp::base::video_info_t' has no member named 'height'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:36:87: error: no matching function for call to 'pbnjson::JObject::JObject(<brace-enclosed initializer list>)'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:36:89: error: no matching function for call to 'pbnjson::JObject::JObject(<brace-enclosed initializer list>)'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:42:43: error: 'const struct gmp::base::audio_info_t' has no member named 'bit_rate'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:43:47: error: 'const struct gmp::base::audio_info_t' has no member named 'sample_rate'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:43:59: error: no matching function for call to 'pbnjson::JObject::JObject(<brace-enclosed initializer list>)'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:51:65: error: 'const struct gmp::base::program_info_t' has no member named 'audio_stream'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:52:65: error: 'const struct gmp::base::program_info_t' has no member named 'video_stream'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:53:31: error: no matching function for call to 'pbnjson::JObject::JObject(<brace-enclosed initializer list>)'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:58:66: error: 'const struct gmp::base::audio_info_t' has no member named 'bit_rate'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:59:69: error: 'const struct gmp::base::audio_info_t' has no member named 'sample_rate'
g-media-pipeline/1.0.0-21.gav.38-r11/git/src/parser/composer.cpp:60:31: error: no matching function for call to 'pbnjson::JObject::JObject(<brace-enclosed initializer list>)'

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-8053] CCC: Various build fixes

Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/g-media-pipeline/+/344542 Fix build with gcc-13]

Change-Id: I838eb3a1875aa3c7c00c9c5d9d8aa5f5f24720d0
---
 src/base/types.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/base/types.h b/src/base/types.h
index 0d6b598..cc893e9 100644
--- a/src/base/types.h
+++ b/src/base/types.h
@@ -17,6 +17,7 @@
 #ifndef SRC_BASE_TYPES_H_
 #define SRC_BASE_TYPES_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
